my_methods.rb

```diff
@@ -0,0 +1,230 @@
+
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    array = self
+    array = array.to_a if array.class == Range
+    array = array.to_a if array.class == Hash
+
+    for item in array
+      yield item
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    array = self
+    array = array.to_a if array.class == Range
+    array = array.to_a if array.class == Hash
+
+    for i in (0...array.length)
+      yield array[i], i
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    array = self
+    array = array.to_a if array.class == Range
+    array = array.values if array.class == Hash
+
+    res = []
+
+    for i in array
+      res.push(i) if yield i
+    end
+    res
+  end
+
+  def my_all?(arg = nil)
+    array = self
+    array = array.to_a if array.class == Range
+    array = array.values if array.class == Hash
+
+    if !block_given? && !arg
+      return false if array.include? false or array.include? nil
+
+      return true
+    end
+
+    return false if array.empty?
+
+    for i in 0...array.length
+      if arg
+        if arg.is_a? Module or arg.is_a? Class
+          return false unless array[i].is_a?(arg)
+        elsif arg.class == Regexp
+          return false if array[i].match(arg).nil?
+        else return false if array[i] != arg
+        end
+      else return false unless yield array[i]
+      end
+    end
+
+    true
+  end
+
+  def my_any?(arg = nil)
+    array = self
+    array = array.to_a if array.class == Range
+    array = array.values if array.class == Hash
+
+    return false if array.empty?
+
+    if !block_given? && !arg
+      return true if array.include?(true)
+
+      res = array.my_all? do |x|
+        !(x != false and !x.nil?)
+      end
+      return false if res
+
+      return true
+    end
+
+    for i in 0...array.length
+      if arg
+        if arg.is_a? Module or arg.is_a? Class
+          return true if array[i].is_a?(arg)
+        elsif arg.class == Regexp
+          return true unless array[i].match(arg).nil?
+        else return true unless array[i] != arg
+        end
+      else return true if yield array[i]
+      end
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    array = self
+    array = array.to_a if array.class == Range
+    array = array.values if array.class == Hash
+
+    if !block_given? && !arg
+      res = array.my_all? do |x|
+        !(x != false and !x.nil?)
+      end
+
+      return true if res
+
+      return false
+    end
+
+    for i in 0...array.length
+      if arg
+        if arg.is_a? Module or arg.is_a? Class
+          return false if array[i].is_a?(arg)
+        elsif arg.class == Regexp
+          return false unless array[i].match(arg).nil?
+        else
+          return false if array[i] == arg
+        end
+      else return false if yield array[i]
+      end
+    end
+
+    true
+  end
```

Comment:
### [OPTIONAL]

- [ ] I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```ruby
def my_none?(arg = nil, &block)
  !my_any?(arg, &block)
end
```

To make this work we have to use block.call(element) instead of yield(element) in my_any?